variables.tf

```
@@ -77,6 +77,25 @@ variable "minio_service" {
   description = "Minio data-object contains service_name, port, access_key and secret_key"
 }
 
+variable "minio_vault_secret" {
+  type = object({
+    use_vault_provider     = bool,
+    vault_kv_policy_name   = string,
+    vault_kv_path          = string,
+    vault_kv_access_key    = string,
+    vault_kv_secret_key    = string
+  })
+  description = "Set of properties to be able to fetch secret from vault"
+  default = {
+    use_vault_provider     = false
+    vault_kv_policy_name   = "kv-secret"
+    vault_kv_path          = "secret/path/to/minio/creds"
+    vault_kv_access_key    = "access_key"
+    vault_kv_secret_key    = "secret_key"
```

I'll leave it with these two suggestions, I guess it would be easier if you include all the necessary changes in one commit
```suggestion
    vault_kv_access_key_name    = "access_key"
    vault_kv_secret_key_name    = "secret_key"
```

---

Issue created. Resolving this 
https://github.com/fredrikhgrelland/terraform-nomad-hive/pull/53#discussion_r520425596